el-patch.el

```diff
@@ -536,38 +547,78 @@ PATCH-DEFINITION is an unquoted list starting with `defun',
   (let ((definition (el-patch--resolve-definition patch-definition t)))
     ;; Then we parse out the definition type and symbol name.
     (cl-destructuring-bind (type name . body) definition
-      (let ((register-patch
-             `(let ((table (or (bound-and-true-p el-patch--patches)
-                               (make-hash-table :test #'eq))))
-                (setq el-patch--patches table)
-                (setq table
-                      (puthash ',name
-                               (gethash
-                                ',name table
-                                (make-hash-table :test #'eq))
-                               table))
-                (setq table
-                      (puthash ',type
-                               (gethash
-                                ',type table
-                                (make-hash-table :test #'eq))
-                               table))
-                (puthash el-patch-variant ',patch-definition table))))
+      (let* ((advise (and el-patch-use-advice
+                          ;; Only advice functions
+                          (let* ((props (alist-get type
+                                                   el-patch-deftype-alist))
+                                 (classifier (plist-get props :classify)))
+                            (and classifier
+                                 (equal
+                                  (caar (funcall classifier definition))
+                                  'function)))
+                          ;; Patches must have the same name and
+                          ;; same number of arguments
```

Comment:
One issue is that there's no way to tell the use that this patch was not made into an advice despite having non-nil `el-patch-use-advice`. I thought about showing a warning, but we would need a way to silence those warnings in init-files.